src/Forum/Content/User.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+/*
+ * This file is part of Flarum.
+ *
+ * (c) Toby Zerner <toby.zerner@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Flarum\Forum\Content;
+
+use Flarum\Api\Client;
+use Flarum\Api\Controller\ShowUserController;
+use Flarum\Frontend\Document;
+use Flarum\Http\Exception\RouteNotFoundException;
+use Flarum\Http\UrlGenerator;
+use Illuminate\Support\Arr;
+use Psr\Http\Message\ServerRequestInterface as Request;
+
+class User
+{
+    /**
+     * @var Client
+     */
+    protected $api;
+
+    /**
+     * @var UrlGenerator
+     */
+    protected $url;
+
+    /**
+     * @param Client $api
+     * @param UrlGenerator $url
+     */
+    public function __construct(Client $api, UrlGenerator $url)
+    {
+        $this->api = $api;
+        $this->url = $url;
+    }
+
+    public function __invoke(Document $document, Request $request)
+    {
+        $queryParams = $request->getQueryParams();
+        $actor = $request->getAttribute('actor');
+        $userId = Arr::get($queryParams, 'username');
+
+        $params = [
+            'id' => $userId,
+        ];
```

Comment:
This doesn't seem to be the same lookup logic as in the client-side variant...

Reply:
The API uses the `id` query param, while the [oute uses `username`.

https://github.com/flarum/core/blob/2f656146a7a47192ce664c3a889349777d92e467/src/Api/Controller/ShowUserController.php#L49-L55

https://github.com/flarum/core/blob/6c96c932e0b6fd5bdf292013f23f4ff590a679bc/src/Forum/routes.php#L30-L33

Reply:
Thanks for the explanation, and the second edition on the phone. :laughing:  This makes sense!